torch/fx/experimental/symbolic_shapes.py

```diff
@@ -7939,6 +7954,21 @@ def constrain_symbol_range(
                 "constrain_symbol_range %s [%s, %s]", s, new_vr.lower, new_vr.upper
             )
 
+    @contextmanager
+    def patch_ra_prelude(self, prelude: sympy.Expr) -> Iterator[None]:
+        """
+        Context manager that ensures all runtime asserts generated while this context manager
+        is active include a prelude expression. This is mainly used in torch.cond to guarantee
+        that runtime asserts in subgraphs are guarded by the original cond predicate, preventing
+        them from leaking into the main graph.
+        """
+        prev = self.ra_prelude
+        self.ra_prelude = prelude
```

Comment:
This needs to 'and' with the previous ra_prelude, doesn't it (i'm thinking about nested torch.cond statements)